library/Core.cpp

```diff
@@ -1489,11 +1489,10 @@ Core::~Core()
     }
     if (d->iothread.joinable())
         con.shutdown();
-    delete d;
 }
 
 Core::Core() :
-    d{new Private},
+    d(new Private),
```

Comment:
Even tough exceptions can't happen here. It is good practice to use make_unique&#60;T>(args...) when initializing unique_ptr. The template function makes sure that allocation is exception safe (not leaking in any scenario). There is forward compatible version in dts namespace. (that is c++14 feature)

Reply:
I was unsure what version of C++ was available so i didnt want to use make_unique. But im glad you have an alternative within dts.